modules/Rock.cmake

```diff
@@ -765,28 +759,85 @@ endfunction()
 # files, they get added to the library and the corresponding header file is
 # passed to moc.
 function(rock_testsuite TARGET_NAME)
+    rock_test_common(${TARGET_NAME} ${ARGN})
+    rock_setup_boost_test(${TARGET_NAME})
+    rock_add_test(${TARGET_NAME})
+endfunction()
+
+## Uses gtest + google-mock as unit testing framework
+# The interface is the same as the one from rock_testsuite
+#
+# Unfortunately, gtest and google-mock do not ship with
+# pre-compiled libraries so we have to add their source files
+# to the test target. If GTEST_DIR or GMOCK_DIR are not set,
+# this function will look for the required source files
+# in /usr/src, which is the default path under ubuntu/debian
+function(rock_gtest TARGET_NAME)
+    if (NOT GTEST_DIR)
+        find_path(GTEST_DIR "src/gtest-all.cc" /usr/src/gtest)
+    endif()
+    if (NOT GMOCK_DIR)
+        find_path(GMOCK_DIR "src/gmock-all.cc" /usr/src/gmock)
+    endif()
+
+    if (NOT GTEST_DIR)
+        message(FATAL_ERROR "Could not find gtest in /usr/src")
+    endif()
+    if (NOT GMOCK_DIR)
+        message(FATAL_ERROR "Could not find google-mock in /usr/src")
+    endif()
+    message(STATUS "gtest found ... building the test suite")
+
+    rock_test_common(${TARGET_NAME} ${GTEST_DIR}/src/gtest-all.cc
+                                    ${GMOCK_DIR}/src/gmock-all.cc
+                                    ${ARGN})
+
+    rock_setup_gtest_test(${TARGET_NAME} ${GMOCK_DIR} ${GTEST_DIR})
+    rock_add_test(${TARGET_NAME})
+endfunction()
+
+macro(rock_setup_gtest_test TARGET_NAME GMOCK_DIR GTEST_DIR)
+    target_include_directories(${TARGET_NAME} SYSTEM PUBLIC ${GMOCK_DIR} ${GTEST_DIR}
```

Comment:
Thanks to you I now realize that target_include_directories was available in cmake 2.8.12 ... I always thought it was 3.0+... Cool !